docs/SDK_v2.md

```diff
@@ -88,3 +91,31 @@ Event -> Message -> bits
 bits -> Message -> Event
 
 bit -> Message -> bits
+
+## Interfaces (Current pre-v2)
+
+```
+Client --> Transport (via TransportBinding) -> Protocol (implements Sender, Receiver)* -> Message Writers
+```
+
+Message Writers read a Message and write what is found into a intermediate
+object that is defined by the [3pl][3pl]. For example, this would ne a nats.Msg
+or http.Request.
+
+Protocol is the thinnest wrapper for the [3pl][3pl] to implement Sender and
+Receiver.
+
+\*Protocol can optionally implement Requester and Responder(not yet defined,
+receiver with reply). Protocol does not want to manage the underlying resources
+that enable the external communication. For example: http.Protocol implements a
+ServeHTTP method which can be leveraged by another SDK component:
+http.Transport, or used directly in custom integrations where more control is
+required.
+
+Transport is legacy. It connects the Client to the Protocol and also manages a
```

Comment:
Transport is the engine that runs the server and handles shutdowns and all that, and we need it for client to be simple. But for protocol to be reusable, we need a it to just handle requests and that is independent of the server setup and/or the server choice (gorilla or something else if they use ServeHTTP).

So maybe this Transport gets renamed to something else and ... I am not totally sure how this will work out yet.

Reply:
Ok in that case, having in this document the "log" of what we have now is fine